planar/planar.go

```diff
@@ -0,0 +1,86 @@
+package planar
+
+import (
+	"math"
+	"math/big"
+)
+
+const Rad = math.Pi / 180
+
+type PointLineDistanceFunc func(line [2][2]float64, point [2]float64) float64
+
+// PerpendicularDistance  provides the distance between a line and a point in Euclidean space.
+func PerpendicularDistance(line [2][2]float64, point [2]float64) float64 {
+
+	deltaX := line[1][0] - line[0][0]
+	deltaY := line[1][1] - line[0][1]
+	denom := math.Abs((deltaY * point[0]) - (deltaX * point[1]) + (line[1][0] * line[0][1]) - (line[1][1] * line[0][0]))
+	num := math.Sqrt(math.Pow(deltaY, 2) + math.Pow(deltaX, 2))
+	if num == 0 {
+		return 0
+	}
+	return denom / num
+}
+
+type Haversine struct {
+	Semimajor *float64
+}
+
+// PerpendicularDistance returns the distance between a point and a line in meters, using the Harversine distance.
+func (hs Haversine) PerpendicularDistance(line [2][2]float64, point [2]float64) float64 {
+	dist := hs.Distance(line[0], point)
+	angle := Angle3Pts(line[0], line[1], point)
+	return math.Abs(dist * math.Sin(angle))
```

Comment:
If I'm reading this right, this seems to be mixing planar operations (Angle3Pts) and spherical operations (hs.Distance). Based on the comment, I think you're looking for [Cross-track distance](https://www.movable-type.co.uk/scripts/latlong.html).

Reply:
Yes, I'm not calculating the Angle3Pts correctly. I should be using a radian_slop function instead of a euclidian slop function. Good catch.